blocks/blockstore/bloom_cache.go

```diff
@@ -25,24 +25,26 @@ func bloomCached(bs Blockstore, ctx context.Context, bloomSize, hashCount int) (
 	bc.total = metrics.NewCtx(ctx, "bloom_total",
 		"Total number of requests to bloom cache").Counter()
 
-	fill := metrics.NewCtx(ctx, "bloom_fill_ratio",
-		"Ratio of bloom filter fullnes, (updated once a minute)").Gauge()
-
 	bc.Invalidate()
 	go bc.Rebuild(ctx)
-	go func() {
-		<-bc.rebuildChan
-		t := time.NewTicker(1 * time.Minute)
-		for {
-			select {
-			case <-ctx.Done():
-				t.Stop()
-				return
-			case <-t.C:
-				fill.Set(bc.bloom.FillRatio())
+	if metrics.Active() {
```

Comment:
- in general: don't use statics. If you run multiple IPFS daemons, the
metrics may clash. If the lib FORCES you to use a global, then add a
guaranteed-unique prefix. Ideally with a local variable that internally
uses the global, threaded through with the context.
- but I don't care enough about this case to force it, if it's way easier
and safe to have thousands of IPFS daemons in the same process, and nothing
will clash, fine.

On Wed, Sep 7, 2016 at 9:10 PM Jeromy Johnson <notifications@github.com>
wrote:

> In blocks/blockstore/bloom_cache.go
> <https://github.com/ipfs/go-ipfs/pull/3189#discussion_r77885221>:
>
> >  	bc.Invalidate()
> >  	go bc.Rebuild(ctx)
> > -	go func() {
> > -		<-bc.rebuildChan
> > -		t := time.NewTicker(1 * time.Minute)
> > -		for {
> > -			select {
> > -			case <-ctx.Done():
> > -				t.Stop()
> > -				return
> > -			case <-t.C:
> > -				fill.Set(bc.bloom.FillRatio())
> > +	if metrics.Active() {
>
> This is still okay, Its just less testable and composable than i was
> hoping.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ipfs/go-ipfs/pull/3189/files/0fa592196ed0f0429ef5a734e4efc17e2f8e0cb9..bb405f316a3819a4dba1d972dca8a547ea80fcb0#r77885221>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAIcoRleigomIjiQWwSzrGwiXL5YAXpYks5qnwxCgaJpZM4J1L0L>
> .
>